src/App.tsx

```diff
@@ -1,14 +1,34 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { SelectedType } from './types/SelectedType';
+
+import { getFilteredData } from './utils/getFilteredData';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
```

Comment:
You're initializing your state with props. This is considered an anti-pattern in React because it can lead to bugs and inconsistencies in your state management. The state should be the single source of truth and should not be dependent on props for its initial value. Instead, consider using a different approach to manage your state. You can read more about this issue [here](https://stackoverflow.com/a/50403930), [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) and [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state).